modules/nf-core/software/snpeff/main.nf

```diff
@@ -0,0 +1,57 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options = initOptions(params.options)
+params.use_cache = false
+params.snpeff_tag = ""
+
+process SNPEFF {
+    label 'process_medium'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda (params.enable_conda ? "bioconda::snpeff=5.0" : null)
+    if (params.use_cache) {
+        if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+            container "https://depot.galaxyproject.org/singularity/snpeff:5.0--hdfd78af_1"
+        } else {
+            container "quay.io/biocontainers/snpeff:5.0--hdfd78af_1"
+        }
+    } else {
+        container "nfcore/snpeff:${params.snpeff_tag}"
+    }
+
+    input:
+    tuple val(meta), path(vcf)
+    val (snpeff_db)
+    path (snpeff_cache)
+
+    output:
+    tuple val(meta), path("*.ann.vcf"), emit: vcf
+    path "*.csv",                       emit: reports
+    path "*.version.txt",               emit: version
+
+    script:
+    def software = getSoftwareName(task.process)
+    def avail_mem = 6
+    if (!task.memory) {
```

Comment:
Maxime and me briefly talked about this. Maybe @drpatelh and @KevinMenden can give some input. I saw this in a couple of modules, but not consistently across all. Is this necessary? All modules have a label, so the task.memory should always be provided. Or is there a use_case where this would not be the case?

Reply:
In this case it is necessary because `snpEff` is a Java tool and hence allocates a tiny amount of memory by default which can break things as reported [here](https://github.com/nf-core/viralrecon/issues/204). Updating the [module](https://github.com/nf-core/viralrecon/blob/master/modules/local/snpeff_ann.nf) sorted the issue.

Reply:
Ah right, I remember this discussion now. We then need to add it to all gatk based modules, I think

Reply:
I'll make a separate PR for that this afternoon then